### PR TITLE
Update StMuMcTrack with num hits to FWD detectors

### PR DESCRIPTION
**Pull request by jdbrice**
Update to StMuMcTrack based on members in g2t_track - specifically add the missing access to number of hits from FWD detectors (and etof, etc.)

### CODE REVIEW

File: StRoot/StMuDSTMaker/COMMON/StMuMcTrack.h [hunk 1]
```diff
@@ -94,7 +104,7 @@ class StMuMcTrack : public TObject {
   Float_t        mpT;          /* Transverse momentum */
   Float_t        mPtot;        /* Total momentum */
   Float_t        mRapidity;    /* Rapidity */
-  ClassDef(StMuMcTrack,1)
+  ClassDef(StMuMcTrack,2)
```
Comment (genevb) [inline StRoot/StMuDSTMaker/COMMON/StMuMcTrack.h:107]: Do any of these changes necessitate an increment of the ClassDef version? ... though I guess it doesn't hurt.

**Comment by plexoos** [comment]
Is the compile error understood?

File: StRoot/StMuDSTMaker/COMMON/StMuMcTrack.h (hunk 1 above)
Comment (jdbrice) [reply StRoot/StMuDSTMaker/COMMON/StMuMcTrack.h:107]: I think since the array is in the middle and it is a fixed length we should bump the version number.

**Comment by jdbrice** [comment]
@plexoos - no idea why the compiler error. It build fine for me in docker container. Will see if I can reproduce

File: StRoot/StMuDSTMaker/COMMON/StMuMcTrack.h (hunk 1 above)
Comment (genevb) [reply StRoot/StMuDSTMaker/COMMON/StMuMcTrack.h:107]: I understand now: `ktot` determines the length of `mHits`, so it has changed. Thanks.

**Comment by jdbrice** [comment]
My fork's main branch was behind star-sw/main - I updated and triggered a new round of tests.

**Comment by jdbrice** [comment]
@plexoos - I wonder if this one also failed due to network issue? I am not sure how to trigger a re-run without a commit. Can you try restarting this one?

**Comment by jdbrice** [comment]
commit [32be173](https://github.com/star-bnl/star-sw/pull/696/commits/32be1736db76a79d32d74c56fb24d06b9020a9e1) fixes the compiler errors - StPicoMcTrack needed updates to mirror changes in MuDst

**Comment by jdbrice** [comment]
Hi @marrbnl and @nigmatkulov - please take a look at this PR now - it is a very small update.

**Review by plexoos — APPROVED** [review]

**Review by akioogawa — APPROVED** [review]

**Pull request merged**